### PR TITLE
chore: remove defaultOnBanReach helper fn

### PR DESCRIPTION
**Pull request by gurgunday**
All custom function parameters, like `onExceeded` and `onExceeding`, only get run if they're defined. However, `onBan` has a default function that is just empty (old, line 281, 268)

Under stress from an IP, that function can get called thousands of times, which probably still wouldn't cause a problem (🤣), but why run it in the first place?

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -265,7 +263,7 @@ function rateLimitRequestHandler (params, pluginComponent) {
 
     if (code === 403) {
       respCtx.ban = true
-      params.onBanReach(req, key)
+      if (typeof params.onBanReach === 'function') params.onBanReach(req, key)
```
Comment (Uzlopak) [inline index.js:266]: Uff. I dont think this is nice. the noop can be optimized by v8, but this is basically a string compare, and is probably slower.

Comment (gurgunday) [reply index.js:266]: I see, I though about function inlining even though that parameter is dynamic, which made me unsure

What about these then? Wouldn't replacing them with empty functions too better then?
```js
      if (typeof params.onExceeding === 'function') {
        params.onExceeding(req, key)
      }
      return
    }
    if (typeof params.onExceeded === 'function') {
      params.onExceeded(req, key)
    }
```

File: index.js [hunk 2]
```diff
@@ -94,23 +91,24 @@ async function fastifyRateLimit (fastify, settings) {
...
   if (typeof settings.errorResponseBuilder === 'function') {
     globalParams.errorResponseBuilder = settings.errorResponseBuilder
     globalParams.isCustomErrorMessage = true
+  } else {
+    globalParams.errorResponseBuilder = defaultErrorResponse
+    globalParams.isCustomErrorMessage = false
   }
 
-  globalParams.skipOnError = settings.skipOnError || false
+  globalParams.skipOnError = typeof settings.skipOnError === 'boolean' ? settings.skipOnError : false
```
Comment (Uzlopak) [inline index.js:106]: if you want to default to false, it is perfectly fine to do `|| false` If you want to default to true, than I would say, this makes sense, but still we would have the "new" approach to do `?? true`.
Or do you want to enforce that skipOnError is definetly a boolean? Why then not do `!!settings.skipOnError'?

File: index.js (hunk 1 above)
Comment (gurgunday) [reply index.js:266]: Let me test actually

File: index.js (hunk 2 above)
Comment (gurgunday) [reply index.js:106]: I did this `!!settings.skipOnError` first, but all other options are type checked, so defaulted to that, just small things

Comment (Uzlopak) [reply index.js:106]: Oh, if it is for consistency, I have no issue with that.

File: index.js (hunk 1 above)
Comment (gurgunday) [reply index.js]: Anyway if Fastify is all for speed, like you said, it shouldn't have typeof checks in request handlers

The types are there for a reason

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by Uzlopak** [comment]
@mcollina 

Are you sure? Is noop not faster than optional chaining?

**Comment by gurgunday** [comment]
I actually forgot about this one, but the main thing is `onExceeding` and `onExceeded` both had `typeof ===` checks while `onBan` called an empty function, which you say is better, which I don't object to

The PR is simply one about consistency, they should either all have empty default functions or be optionally called — like how either all options should be type checked or none

PS: I think the difference is extremely small

**Comment by Uzlopak** [comment]
What would be more interesting would be a benchmark

**Comment by gurgunday** [comment]
Let me try that even though it's likely that the difference will be hard to measure, especially between an empty function call and optional chaining

**Comment by gurgunday** [comment]
No medians, no statistics, just the highest scores I could get after running each of them 10 times. Obviously not a breakthrough improvement, and the difference between the last 2 is definitely non-significant IMO

My intention here was to have the same way of handling defaults within the codebase

master, node 20
```shell
150 connections


┌─────────┬──────┬──────┬───────┬──────┬────────┬─────────┬────────┐
│ Stat    │ 2.5% │ 50%  │ 97.5% │ 99%  │ Avg    │ Stdev   │ Max    │
├─────────┼──────┼──────┼───────┼──────┼────────┼─────────┼────────┤
│ Latency │ 2 ms │ 2 ms │ 3 ms  │ 5 ms │ 2.2 ms │ 1.57 ms │ 224 ms │
└─────────┴──────┴──────┴───────┴──────┴────────┴─────────┴────────┘
┌───────────┬─────────┬─────────┬─────────┬─────────┬──────────┬─────────┬─────────┐
│ Stat      │ 1%      │ 2.5%    │ 50%     │ 97.5%   │ Avg      │ Stdev   │ Min     │
├───────────┼─────────┼─────────┼─────────┼─────────┼──────────┼─────────┼─────────┤
│ Req/Sec   │ 47839   │ 47839   │ 51551   │ 51903   │ 51073.46 │ 1245.38 │ 47825   │
├───────────┼─────────┼─────────┼─────────┼─────────┼──────────┼─────────┼─────────┤
│ Bytes/Sec │ 16.8 MB │ 16.8 MB │ 18.7 MB │ 18.8 MB │ 18.5 MB  │ 590 kB  │ 16.8 MB │
└───────────┴─────────┴─────────┴─────────┴─────────┴──────────┴─────────┴─────────┘

Req/Bytes counts sampled once per second.
# of samples: 11

5000 2xx responses, 556809 non 2xx responses
562k requests in 11.01s, 203 MB read
```

PR, node 20
```shell
150 connections


┌─────────┬──────┬──────┬───────┬──────┬─────────┬─────────┬────────┐
│ Stat    │ 2.5% │ 50%  │ 97.5% │ 99%  │ Avg     │ Stdev   │ Max    │
├─────────┼──────┼──────┼───────┼──────┼─────────┼─────────┼────────┤
│ Latency │ 2 ms │ 2 ms │ 3 ms  │ 5 ms │ 2.16 ms │ 2.12 ms │ 269 ms │
└─────────┴──────┴──────┴───────┴──────┴─────────┴─────────┴────────┘
┌───────────┬───────┬───────┬─────────┬─────────┬──────────┬─────────┬───────┐
│ Stat      │ 1%    │ 2.5%  │ 50%     │ 97.5%   │ Avg      │ Stdev   │ Min   │
├───────────┼───────┼───────┼─────────┼─────────┼──────────┼─────────┼───────┤
│ Req/Sec   │ 48351 │ 48351 │ 52543   │ 52959   │ 51780.37 │ 1455.55 │ 48325 │
├───────────┼───────┼───────┼─────────┼─────────┼──────────┼─────────┼───────┤
│ Bytes/Sec │ 17 MB │ 17 MB │ 19.1 MB │ 19.2 MB │ 18.7 MB  │ 659 kB  │ 17 MB │
└───────────┴───────┴───────┴─────────┴─────────┴──────────┴─────────┴───────┘

Req/Bytes counts sampled once per second.
# of samples: 11

5000 2xx responses, 564578 non 2xx responses
570k requests in 11.02s, 206 MB read
```

PR with empty default functions, node 20
```shell
150 connections


┌─────────┬──────┬──────┬───────┬──────┬─────────┬─────────┬────────┐
│ Stat    │ 2.5% │ 50%  │ 97.5% │ 99%  │ Avg     │ Stdev   │ Max    │
├─────────┼──────┼──────┼───────┼──────┼─────────┼─────────┼────────┤
│ Latency │ 2 ms │ 2 ms │ 3 ms  │ 5 ms │ 2.17 ms │ 2.13 ms │ 273 ms │
└─────────┴──────┴──────┴───────┴──────┴─────────┴─────────┴────────┘
┌───────────┬─────────┬─────────┬───────┬─────────┬──────────┬─────────┬─────────┐
│ Stat      │ 1%      │ 2.5%    │ 50%   │ 97.5%   │ Avg      │ Stdev   │ Min     │
├───────────┼─────────┼─────────┼───────┼─────────┼──────────┼─────────┼─────────┤
│ Req/Sec   │ 47903   │ 47903   │ 52351 │ 52703   │ 51640.73 │ 1499.95 │ 47898   │
├───────────┼─────────┼─────────┼───────┼─────────┼──────────┼─────────┼─────────┤
│ Bytes/Sec │ 16.8 MB │ 16.8 MB │ 19 MB │ 19.1 MB │ 18.7 MB  │ 679 kB  │ 16.8 MB │
└───────────┴─────────┴─────────┴───────┴─────────┴──────────┴─────────┴─────────┘

Req/Bytes counts sampled once per second.
# of samples: 11

5000 2xx responses, 563085 non 2xx responses
568k requests in 11.02s, 206 MB read
```


Code
```js
import Fastify from 'fastify'
import rateLimit from '../index.js'

const fastify = Fastify()
await fastify.register(rateLimit, { max: 5000, timeWindow: '5h', ban: 1 })

fastify.get('/', async (req, reply) => 'hello!')

fastify.listen({ host: '127.0.0.1', port: 5001 }, (err, address) => {
  if (err) throw err
  console.log(address)
})
```

**Review by Uzlopak — APPROVED** [review]

**Pull request merged**